liblocation/src/main/java/com/mapbox/android/core/location/GoogleLocationEngine.java

```diff
@@ -102,17 +109,28 @@ public void onConnectionSuspended(int cause) {
   public void onConnectionFailed(@NonNull ConnectionResult connectionResult) {
   }
 
+  @SuppressLint("MissingPermission")
   @Override
   public Location getLastLocation() {
     if (googleApiClient.isConnected()) {
+      final Location[] lastLocation = {null};
+
       //noinspection MissingPermission
-      return LocationServices.FusedLocationApi.getLastLocation(googleApiClient);
+      fusedLocationProviderClient.getLastLocation().addOnSuccessListener(new OnSuccessListener<Location>() {
+        @Override
+        public void onSuccess(Location location) {
+          lastLocation[0] = location;
+        }
+      });
+
+      return lastLocation[0];
     }
 
     //noinspection MissingPermission
     return null;
   }
 
+  @SuppressLint("MissingPermission")
```

Comment:
instead of suppressing Lint can we add the required permissions annotations instead?